src/m_fixed.h

```diff
@@ -38,8 +38,8 @@
       {
           return a / b;
       }
-      int32_t lo = a;
-      int32_t hi = a >> 32;
+      volatile int32_t lo = a;
```

Comment:
How about `int32_t lo = (int32_t) a;`?
It seems to be another case of UB that will differ in various versions of compilers.

Reply:
Tried this, same crash.

```patch
--- a/src/m_fixed.h
+++ b/src/m_fixed.h
@@ -38,8 +38,14 @@
       {
           return a / b;
       }
-      int32_t lo = a;
-      int32_t hi = a >> 32;
+      union {
+          int64_t v64;
+          int32_t v32[2];
+      } d = {a};
+
+      int32_t lo = d.v32[0];
+      int32_t hi = d.v32[1];
+
       asm("idivl %[divisor]" : "+a" (lo), "+d" (hi) : [divisor] "r" (b));
       return lo;
   }
```

Reply:
> How about `int32_t lo = (int32_t) a;`?

Same crash.

Reply:
I think the problem is that `a` and `b` are both input and output variables to the `asm()` call. However, the issue is very compiler specific. Ubuntu-latest uses gcc-11.2 and I cannot reproduce it with gcc-14.2 on my main system with the exact same build flags. Also, it is optimization specific and won't occur with gcc-11.2 in a debug build with `-O0`.

Reply:
Reminds me of this bug, where we also never really understood what was going on:

https://github.com/fabiangreffrath/woof/commit/3c526e70688b18f65568ef3b3ee235fd3a9c2c13

Reply:
This was a miscompilation with Clang -O3. May be fixed in newer versions of the compiler.

Reply:
Yes, it's possible the same thing happens here with gcc-11.

Reply:
Anyway, I suggest to remove the GCC version of `div64_32`. It helps only for old CPUs and is only noticeable with voxels (we have very few `FixedDiv` in other code).